core/chains/evm/log/orm.go

```diff
@@ -110,6 +113,18 @@ func (o *orm) MarkBroadcastConsumed(blockHash common.Hash, blockNumber uint64, l
 	return errors.Wrap(err, "failed to mark log broadcast as consumed")
 }
 
+// MarkBroadcastsUnconsumed implements the ORM interface.
+func (o *orm) MarkBroadcastsUnconsumed(fromBlock int64, qopts ...pg.QOpt) error {
+	q := o.q.WithOpts(qopts...)
+	err := q.ExecQ(`
+        UPDATE log_broadcasts
+        SET consumed = false
+        WHERE block_number >= $1
+		AND evm_chain_id = $2
```

Comment:
Is there already an index for this?

Reply:
I suspect it's important to add. Also, I'm unsure if it matters or would help to flip the ordering to `evm_chain_id AND block_number` :thinking: - I've been in NoSQL land for too long...

Reply:
you would want to put the one that filters the most results first. as that would speed things up.

also at a quick glance i dont think those indexes would get used here so seems you would want a new one

Reply:
Good catch @jmank88 .

Ordering in the query doesn't matter, but this query is unindexed currently.

@Nic0S postgres cannot (easily) use isolated indexes if they are not at the leftmost side of the multi-column index.

You will need an additional index.

`CREATE INDEX idx_log_broadcasts_block_number_evm_chain_id ON log_broadcasts (evm_chain_id, block_number)`

Unfortunately this index will be enormous. Is there any way to rewrite the query to be more efficient?

Reply:
Wait, are we sure there isn't an index already? FindBroadcasts does this:
```go
func (o *orm) FindBroadcasts(fromBlockNum int64, toBlockNum int64) ([]LogBroadcast, error) {
	var broadcasts []LogBroadcast
	query := `
		SELECT block_hash, consumed, log_index, job_id FROM log_broadcasts
		WHERE block_number >= $1
		AND block_number <= $2
		AND evm_chain_id = $3
	`
```

Reply:
Want to note that this query will only ever be run as part of a manual log replay, not as part of normal operation. Even if it hung for several seconds I don't think it would cause too much of an issue.

Reply:
I guess it could be worse than several seconds though, really not sure

Reply:
TIL about only taking advantage of the leftmost side of a compound index, not sure why I thought it was rightmost.

Reply:
Looks like `FindBroadcasts` has never been properly indexed 😱 .

Yes, we should have the index I listed for sure then.

Reply:
Done!